src/coap_gnutls.c

```diff
@@ -620,14 +669,17 @@ static int cert_verify_gnutls(gnutls_session_t g_session)
                 (coap_session_t *)gnutls_transport_get_ptr(g_session);
   coap_gnutls_context_t *g_context =
              (coap_gnutls_context_t *)c_session->context->dtls_context;
-  char *cn = NULL;
   int alert = GNUTLS_A_BAD_CERTIFICATE;
   int ret;
+  coap_gnutls_certificate_info_t cert_info;
+  gnutls_certificate_type_t cert_type;
+
+  cert_info.san_or_cn = NULL;
```

Comment:
I would prefer to clear the cert_info structure in `get_san_or_cn()`.

Reply:
Done